x/distribution/keeper/allocation.go

```diff
@@ -0,0 +1,63 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/distribution/types"
+)
+
+// Allocate fees handles distribution of the collected fees
+func (k Keeper) AllocateFees(ctx sdk.Context) {
+	ctx.Logger().With("module", "x/distribution").Error(fmt.Sprintf("allocation height: %v", ctx.BlockHeight()))
+
+	// if there is no power in the system nothing should be allocated
+	bondedTokens := k.stakeKeeper.TotalPower(ctx).TruncateInt()
+	if bondedTokens.IsZero() {
+		return
+	}
+
+	// get the proposer of this block
+	proposerConsAddr := k.GetProposerConsAddr(ctx)
+	proposerValidator := k.stakeKeeper.ValidatorByConsAddr(ctx, proposerConsAddr)
+	proposerDist := k.GetValidatorDistInfo(ctx, proposerValidator.GetOperator())
+
+	// get the fees which have been getting collected through all the
+	// transactions in the block
+	feesCollected := k.feeCollectionKeeper.GetCollectedFees(ctx)
+	feesCollectedDec := types.NewDecCoins(feesCollected)
+
+	// allocated rewards to proposer
+	sumPowerPrecommitValidators := k.GetSumPrecommitPower(ctx)
+	percentVoting := sdk.NewDec(sumPowerPrecommitValidators).QuoInt(bondedTokens)
```

Comment:
I think this is subtly wrong, it's off by two blocks. The stake state reflects the *current* validator set's total power, but the validator set which could have signed the precommit only had the total power of two blocks ago. Maybe we should update `TotalPower` to include this offset - or maybe we are OK with the slight inaccuracy, but at least worth noting.

Reply:
oh duhdoy - we should just sum it up from the beginblock as well :) - shouldn't really add much overhead and then it's fully correct